gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java

```diff
@@ -0,0 +1,462 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import kafka.api.PartitionFetchInfo;
+import kafka.api.PartitionOffsetRequestInfo;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.FetchRequest;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.OffsetResponse;
+import kafka.javaapi.PartitionMetadata;
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.javaapi.message.ByteBufferMessageSet;
+import kafka.message.MessageAndOffset;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+
+/**
+ * Wrapper class that contains two alternative Kakfa APIs: an old low-level Scala-based API, and a new API.
+ * The new API has not been implemented since it's not ready to be open sourced.
+ *
+ * @author ziliu
+ */
+public class KafkaWrapper implements Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaWrapper.class);
+
+  private static final String USE_NEW_KAFKA_API = "use.new.kafka.api";
+  private static final boolean DEFAULT_USE_NEW_KAFKA_API = false;
+  private static final String KAFKA_BROKERS = "kafka.brokers";
+
+  private final List<String> brokers;
+  private final KafkaAPI kafkaAPI;
+
+  private final boolean useNewKafkaAPI;
+
+  private static class Builder {
+    private boolean useNewKafkaAPI = DEFAULT_USE_NEW_KAFKA_API;
+    private List<String> brokers = Lists.newArrayList();
+
+    private Builder withNewKafkaAPI() {
+      this.useNewKafkaAPI = true;
+      return this;
+    }
+
+    private Builder withBrokers(List<String> brokers) {
+      for (String broker : brokers) {
+        Preconditions.checkArgument(broker.matches(".+:\\d+"),
+            String.format("Invalid broker: %s. Must be in the format of address:port.", broker));
+      }
+      this.brokers = Lists.newArrayList(brokers);
+      return this;
+    }
+
+    private KafkaWrapper build() {
+      Preconditions.checkArgument(!brokers.isEmpty(), "Need to specify at least one Kafka broker.");
+      return new KafkaWrapper(this);
+    }
+  }
+
+  private KafkaWrapper(Builder builder) {
+    this.useNewKafkaAPI = builder.useNewKafkaAPI;
+    this.brokers = builder.brokers;
+    this.kafkaAPI = getKafkaAPI();
+  }
+
+  /**
+   * Create a KafkaWrapper based on the given type of Kafka API and list of Kafka brokers.
+   *
+   * @param state A {@link State} object that should contain a list of comma separated Kafka brokers.
+   * It may optionally specify whether to use the new Kafka API.
+   */
+  public static KafkaWrapper create(State state) {
```

Comment:
Can you add more information to the javadocs to describe what parameters this method expects in the "state" parameter? e.g. KAFKA_BROKERS, USER_NEW_KAFKA_API, KAFKA_BROKERS, etc.